armi/reactor/converters/axialExpansionChanger.py

```diff
@@ -0,0 +1,492 @@
+                    if growFrac >= 0.0:
+                        c.height = (1.0 + growFrac) * b.p.height
+                    else:
+                        c.height = (1.0 / (1.0 - growFrac)) * b.p.height
+                    # align linked components
+                    if ib == 0:
+                        c.zbottom = 0.0
+                    else:
+                        if self.linked.linkedComponents[c][0] is not None:
+                            # use linked components below
+                            c.zbottom = self.linked.linkedComponents[c][0].ztop
+                        else:
+                            # otherwise there aren't any linked components
+                            # so just set the bottom of the component to
+                            # the top of the block below it
+                            c.zbottom = self.linked.linkedBlocks[b][0].p.ztop
+                    c.ztop = c.zbottom + c.height
+                    # redistribute block boundaries if on the target component
+                    if self.expansionData.isTargetComponent(c):
+                        b.p.ztop = c.ztop
+
+            ## see also b.setHeight()
+            # - the above not chosen due to call to calculateZCoords
+            oldComponentVolumes = [c.getVolume() for c in b]
+            oldHeight = b.getHeight()
+            b.p.height = b.p.ztop - b.p.zbottom
+            _checkBlockHeight(b)
+            _conserveComponentMass(b, oldHeight, oldComponentVolumes)
+            ## set block mid point and redo mesh
+            # - functionality based on assembly.calculateZCoords()
+            b.p.z = b.p.zbottom + b.p.height / 2.0
+            mesh.append(b.p.ztop)
+            b.spatialLocator = self.linked.a.spatialGrid[0, 0, ib]
+
+        bounds = list(self.linked.a.spatialGrid._bounds)
+        bounds[2] = array(mesh)
+        self.linked.a.spatialGrid._bounds = tuple(bounds)
+
+    def axiallyExpandCoreThermal(self, r, tempGrid, tempField):
+        """
+        Perform thermally driven axial expansion of the core.
+
+        Parameters
+        ----------
+        r
+            ARMI reactor to be expanded
+        tempGrid : dictionary
+            keys --> assembly object
+            values --> grid
+        tempField : dictionary
+            keys --> assembly object
+            values --> temperatures
+
+        """
+        for a in r.core.getAssemblies(includeBolAssems=True):
+            self.setAssembly(a)
+            self.expansionData.mapHotTempToComponents(tempGrid[a], tempField[a])
+            self.expansionData.computeThermalExpansionFactors()
+            self.axiallyExpandAssembly()
+
+        self._manageCoreMesh(r)
+
+    def axiallyExpandCorePercent(self, r, components, percents):
+        """
+        Perform axial expansion of the core driven by user-defined expansion percentages.
+
+        Parameters
+        ----------
+        r
+            ARMI reactor to be expanded
+        components : dictionary
```

Comment:
I believe the type should just be `dict` for Sphinx docs to pick this up correctly rather than writing out `dictionary`.

Reply:
Addressed.

https://github.com/terrapower/armi/pull/606/commits/b35802979a71dbb8f69c0b66baee1ae8bad9fa35